### ISSUE TITLE
Project-wide execution directory not respected

### ISSUE THREAD

**Issue by pat-alt** [PumasAI/QuartoNotebookRunner.jl#182]
This might be related to https://github.com/quarto-dev/quarto-cli/issues/3157: I've noted that specifying the project-wide execution directory has no effect when using the `julia` engine. 

### Example

I've added a reproducible example [here](https://github.com/pat-alt/quarto-example/blob/main/julia/execute_dir.qmd). You'll notice that in [_quarto.yml](https://github.com/pat-alt/quarto-example/blob/main/_quarto.yml) I specified the project directory as the execution directory. Doing so with the Jupyter kernel for Julia sets the working directory to the project root as expected, but not when using the `julia` engine.

### Motivation

In my experience, setting the execution directory to the project root makes it easier to handle large projects (e.g. when generating package documentation using both Quarto and Documenter.jl). The reason is that in this scenario every `.qmd` starts from the same working directory, independent of where in the project the file is placed. 

### Ideal scenario

Ideally, the execution flags would also respect this global option, i.e. if I've specified `execute-dir: project` in my global `_quarto.yml` then I can specify the project directory relative to that. Suppose, for example, I have a special environment in my `docs/` folder that I want to use as the default environment for all of my `.qmd` files inside `docs/src`. Then I would like to be able to simply specify `exeflags: ["--project=docs/"]` for all of them (so `docs/` relative to the root directory as opposed to `../**/docs/` relative to the notebook).

**Comment by MichaelHatherly** [comment]
Yes, closely related to that upstream issue. What we can do here is support `execute-dir` within the frontmatter we do receive from the `quarto` process and execute in the directory provided.

**Comment by asinghvi17** [comment]
For what it's worth, even when I supply `execute-dir` in the Quarto front matter on each page, it doesn't seem to apply (using the Julia engine) either.

**Comment by MichaelHatherly** [comment]
Correct, `execute-dir` just hasn't been implemented yet anywhere in this package. Probably a relatively straightforward addition though.
